### PR TITLE
feat: Revamp No-Code Action Form

### PR DESCRIPTION
**Pull request by gupta-piyush19**
⚠️ Data migration: `pnpm data-migration:extended-noCodeActions `

@mattinannt

## What does this PR do?
- Revamp No-Code Action Form
- Automatic actions except New sessions are of type noCodeActions

Fixes 
https://github.com/formbricks/internal/issues/193

## How should this be tested?

- Create new actions using the new form
- Test the new actions in the demo app, with new specific URL support for all the noCode actions.

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by pandeymangg — CHANGES_REQUESTED** [review]
Thanks for the changes @gupta-piyush19 looks great 🚀 
Please change the no code action logic in the js package to use common functions like earlier, I already looked at the data migration and it looks good as well, did some changes in the react hook form implementation as well, please do this and this will be ready after a final round of testing 😊

**Review by pandeymangg — APPROVED** [review]
LGTM! 🚀

**Review by mattinannt — CHANGES_REQUESTED** [review]

File: packages/database/package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "data-migration:userId": "ts-node ./data-migrations/20240408123456_userid_migration/data-migration.ts",
     "data-migration:refactor-actions": "ts-node ./data-migrations/20240501111944_refactors_actions_and_removes_inline_triggers/data-migration.ts",
     "data-migration:mls-welcomeCard-fix": "ts-node ./data-migrations/20240318050527_add_languages_and_survey_languages/data-migration-welcomeCard-fix.ts",
-    "data-migration:v2.0": "pnpm data-migration:mls && pnpm data-migration:styling && pnpm data-migration:styling-fix && pnpm data-migration:website-surveys && pnpm data-migration:userId && pnpm data-migration:mls-welcomeCard-fix && pnpm data-migration:refactor-actions"
+    "data-migration:v2.0": "pnpm data-migration:mls && pnpm data-migration:styling && pnpm data-migration:styling-fix && pnpm data-migration:website-surveys && pnpm data-migration:userId && pnpm data-migration:mls-welcomeCard-fix && pnpm data-migration:refactor-actions",
+    "data-migration:extended-noCode-Actions": "ts-node ./data-migrations/20240524053239_extends_no_code_action_schema/data-migration.ts"
```
[the remark this answers is not in the document]
Comment (mattinannt) [reply packages/database/package.json]: solved :-)

**Review by mattinannt — APPROVED** [review]
@gupta-piyush19 thanks for making the changes; works great! 😊🚀
Ready to be merged :-)

**Pull request merged**